src/app/applications/multicast/multicast-edit/multicast-edit.component.ts

```diff
@@ -0,0 +1,188 @@
+    private router: Router,
+    public multicastService: MulticastService,
+    public errorMessageService: ErrorMessageService,
+    public scrollToTopService: ScrollToTopService,
+    public snackService: SnackService,
+    public applicationService: ApplicationService
+  ) {}
+
+  ngOnInit(): void {
+    this.translate
+      .get([
+        'FORM.CREATE-NEW-MULTICAST',
+        'FORM.EDIT-MULTICAST',
+        'MULTICAST.SAVE',
+        'NAV.MULTICAST',
+        'GEN.BACK',
+      ])
+      .subscribe((translations) => {
+        this.multicastId = +this.route.snapshot.paramMap.get('multicastId');
+        this.applicationId = +this.route.snapshot.paramMap.get('id');
+
+        if (this.multicastId) {
+          this.title = translations['FORM.EDIT-MULTICAST'];
+        } else {
+          this.title = translations['FORM.CREATE-NEW-MULTICAST'];
+        }
+        this.submitButton = translations['MULTICAST.SAVE'];
+        this.backButtonTitle = translations['GEN.BACK'];
+      });
+
+    this.getApplication(this.applicationId);
+
+    if (this.multicastId) {
+      // If edit is pressed, then get the specific multicast.
+      this.getMulticast(this.multicastId);
+    }
+  }
+  onSubmit(): void {
+    if (this.multicastId) {
+      // if already created, only update
+      this.updateMulticast();
+    } else {
+      // else create new
+      this.createMulticast();
+    }
+  }
+
+  getMulticast(id: number) {
+    this.multicastSubscription = this.multicastService
+      .get(id)
+      .subscribe((response: Multicast) => {
+        this.multicast = response; // gets the multicast and set's local multicast. Used when update.
+      });
+  }
+
+  getApplication(id: number) {
+    this.applicationService
+      .getApplication(id)
+      .subscribe((application) => (this.iotDevices = application.iotDevices));
+  }
+
+  //only if classB can be used
+  // showPeriodicity(): boolean {
+  //   if (this.multicast.groupType === MulticastType.ClassB) {
+  //     return true;
+  //   } else return false;
+  // }
+
+  updateMulticast(): void {
+    this.resetErrors();
+    this.multicast.applicationID = this.applicationId;
+
+    this.multicastService.update(this.multicast).subscribe(
+      () => {
+        this.showUpdatedSnack();
+        this.routeBack();
+      },
+      (error: HttpErrorResponse) => {
+        this.showFailSnack();
+        this.handleError(error);
+        this.formFailedSubmit = true;
+      }
+    );
+  }
+  createMulticast(): void {
+    this.resetErrors();
+    this.multicast.applicationID = this.applicationId;
+
+    this.multicastService.create(this.multicast).subscribe(
+      () => {
+        this.showSavedSnack();
+        this.routeBack();
+      },
+      (error: HttpErrorResponse) => {
+        this.showFailSnack();
+        this.handleError(error);
+        this.formFailedSubmit = true;
+      }
+    );
+  }
+  public compare(o1: any, o2: any): boolean {
+    return o1 && o2 && o1.id == o2.id;
```

Comment:
The first two checks are for whether they're defined. Can the list contain undefined/null?